src/OrbitBase/ThreadPool.cpp

```diff
@@ -19,16 +20,27 @@
 namespace orbit_base {
 namespace {
 
+ABSL_CONST_INIT absl::Mutex g_default_thread_pool_mutex(absl::kConstInit);
+ABSL_CONST_INIT std::shared_ptr<ThreadPool> g_default_thread_pool
+    ABSL_GUARDED_BY(g_default_thread_pool_mutex);
+ABSL_CONST_INIT std::shared_ptr<ThreadPool> g_user_default_thread_pool
```

Comment:
I think this is just copying the behaviour that we have internally.

Reply:
Removed "g_user_default_thread_pool". There is only "g_default_thread_pool" now, which is either one that was set by the user, or ThreadPool's internally created one. Made the interface stricter, we can still call "SetDefaultThreadPool(...)", but only once, with a non-null thread pool, and before any call to "GetDefaultThreadPool()". This makes it so that there is only ever one default thread pool created and used.